libcloudforensics/providers/gcp/internal/storagetransfer.py

```diff
@@ -0,0 +1,163 @@
+
+    Args:
+      s3_path (str): File path to the S3 resource.
+          Ex: s3://test/bucket/obj
+      zone (str): The AWS zone in which resources are located.
+        Available zones are listed at:
+        https://cloud.google.com/storage-transfer/docs/create-manage-transfer-program#s3-to-cloud  # pylint: disable=line-too-long
+      gcs_path (str): File path to the target GCS bucket.
+          Ex: gs://bucket/folder
+    Returns:
+      Dict: An API operation object for a Google Cloud Storage Transfer operation.
+        https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferOperations/list  # pylint: disable=line-too-long
+    Raises:
+      TransferCreationError: If the transfer couldn't be created.
+      TransferExecutionError: If the transfer couldn't be run.
+    """
+    aws_creds = account.AWSAccount(zone).session.get_credentials()
+    s3_bucket, s3_path = SplitStoragePath(s3_path)
+    gcs_bucket, gcs_path = SplitStoragePath(gcs_path)
+    if not gcs_path.endswith('/'):
+      gcs_path = gcs_path + '/'
+    # Don't specify a path if we're writing to the bucket root.
+    if gcs_path == '/':
+      gcs_path = ''
+    today = datetime.datetime.now()
+    transfer_job_body = {
+        'projectId': self.project_id,
+        'description': 'created_by_cfu',
+        'transferSpec': {
+            'objectConditions': {
+                'includePrefixes': [s3_path]
+            },
+            'awsS3DataSource': {
+                'bucketName': s3_bucket,
+                'awsAccessKey': {
+                    'accessKeyId': aws_creds.access_key,
+                    'secretAccessKey': aws_creds.secret_key
+                }
+            },
+            'gcsDataSink': {
+                'bucketName': gcs_bucket, 'path': gcs_path
+            }
+        },
+        'schedule': {
+            'scheduleStartDate': {
+                'year': today.year, 'month': today.month, 'day': today.day
+            },
+            'scheduleEndDate': {
+                'year': today.year, 'month': today.month, 'day': today.day
+            },
+            'endTimeOfDay': {}
+        },
+        'status': 'ENABLED'
+    }
+    logger.info('Creating transfer job')
+    gcst_jobs = self.GcstApi().transferJobs()
+    create_request = gcst_jobs.create(body=transfer_job_body)
+    transfer_job = create_request.execute()
+    logger.info('Job created: {0:s}'.format(str(transfer_job)))
+    job_name = transfer_job.get('name', None)
+    if job_name is None:
+      raise errors.TransferCreationError(
+          'Could not create transfer. Job output: {0:s}'.format(
+              str(transfer_job)),
+          __name__)
+    logger.info('Job created: {0:s}'.format(job_name))
+    gcst_transfers = self.GcstApi().transferOperations()
+    filter_string = ('{{"projectId": "{0:s}", "jobNames": ["{1:s}"]}}').format(
+        self.project_id, job_name)
+    status = {}
+    while 'operations' not in status:
+      time.sleep(5)
```

Comment:
Possible future enhancement - Dynamically calculate the expected time, and adjust timeouts accordingly. Similar logic exists in  the snapshot copy operation here[0]

[0] https://github.com/google/cloud-forensics-utils/blob/main/libcloudforensics/providers/aws/forensics.py#L416

Reply:
Nice to have in a future CL :)